tests/Unit/BankServiceTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace StdGroup\Tests\Unit;
+use Carbon\Carbon;
+use PHPUnit\Framework\TestCase;
+use StdGroup\App\BankService;
+use StdGroup\App\Entity\Card;
+
+
+class BankServiceTest extends  TestCase
+{
+    public function testCalculateFeeWithVipCard() {
+
+        // id, isVip, time, expectResult
+        $cases = [
+            [1, 2, '', 0], //isVip
+            [0, 0, '2022-04-30 08:45:05', 110], // holidays
+            [0, 0, '2022-07-10 08:45:05', 110], // weekends
+            [0, 0, '2022-07-11 06:43:05', 0], // not VIP && not holidays and weekends && time free 00:00 - 08:45
+            [0, 0, '2022-07-12 23:43:05', 110], // not VIP && not weekends and holidays && out time free
+        ];
+
+        foreach ($cases as $case) {
+            $bankService = new BankService();
+            $card = new Card($case[0], $case[1]);
+            Carbon::setTestNow($case[2]);
+            $result = $bankService->calculateFee($card);
+            $this->assertEquals($result, $case[3]);
+        }
+    }
```

Comment:
Tách case ra nha e, ko ai gộp cases lại rồi foreach như này cả. Test case cần phải đơn giản, ko nên chứ logic, những case nào trùng lặp thì có thể dùng [`@dataProvider`](https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#data-providers)

Mỗi case test cái gì thì đặt tên cụ thể test case ra nhé, ví dụ `test_it_should_return_no_fee_when_card_is_vip`

:point_right: Checklist https://sun-unit-test-training.github.io/php-testing-guideline/06-checklist/

Reply:
okee anh ạ. Em hiểu r